packages/service-core/src/storage/ReportStorage.ts

```diff
@@ -0,0 +1,41 @@
+import { event_types } from '@powersync/service-types';
+
+/**
+ * Represents a configured report storage.
+ *
+ * Report storage is used for storing localized data for the instance.
+ * Data can then be used for reporting purposes.
+ *
+ */
+export interface ReportStorage extends AsyncDisposable {
+  /**
+   * Report a client connection.
+   */
+  reportClientConnection(data: event_types.ClientConnectionBucketData): Promise<void>;
+  /**
+   * Report a client disconnection.
+   */
+  reportClientDisconnection(data: event_types.ClientDisconnectionEventData): Promise<void>;
+  /**
+   * Get currently connected clients.
+   * This will return any short or long term connected clients.
+   * Clients that have no disconnected_at timestamp and that have a valid jwt_exp timestamp are considered connected.
+   * @param data.range.start Needs to be UTC time string
+   * @param data.range.end Optional needs to be UTC time string
+   */
+  getConnectedClients(data: event_types.ClientConnectionsRequest): Promise<event_types.ClientConnectionReportResponse>;
```

Comment:
This helps, but still need more detail here.

The description says "currently connected clients", but the parameters allow passing in a timestamp range. So for example if pass in `{start: '2025-08-01T00:00Z', end: '2025-08-01T00:00Z'}`, would I get a list of clients who were connected at that time? Or in which cases should the time range be used?